schema/cosa/cosa_v1.go

```diff
@@ -103,7 +103,8 @@ type BuildArtifacts struct {
 
 type Cloudartifact struct {
 	Bucket string `json:"bucket,omitempty"`
-	Image  string `json:"image"`
+	Image  string `json:"image,omitempty"`
+	Object string `json:"object,omitempty"`
```

Comment:
@jlebon @cgwalters - I suggested @Prashanth684 add an `object` field here so they can identify things via `bucket`/`object` rather than re-use the `image` key here (which typically implies bootable image (like AMI)).

Reply:
Hmm, feels like we should just get rid of the generic `Cloudartifact`, and have separate `AzureArtifact`, `IbmCloudArtifact`, and `PowerVirtualServerArtifacts` types. WDYT?

Reply:
yeah, he previously had something like `IbmCloudArtifact` to represent both IBMCloud and PowerVS but I suggested he use CloudArtifact (at the time we hadn't proposed adding the new `object` field). I honestly think we don't really need that. Feels like re-using cloudArtifact here should be fine since it shares concepts with other clouds (unlike GCP, which has `projects` and image families and such).

Reply:
@jlebon thoughts?

Reply:
So let me make sure I understand:
- `Azure` uses only `Image` and `URL`
- `IbmCloud` uses `Bucket`, `Object`, `Region`, and `Url`
- `PowerVirtualServer` uses `Bucket` and `Object`

?

I mean, it depends how "well-typed" we want to be. Those clearly could use separate types, and then validation would be that much stronger (which means e.g. we don't run the risk of pushing an `azure` key with fields which don't make sense). While in FCOS these details are mostly internal, because we have layers of metadata on top with very different schemas, in RHCOS the `meta.json` is only lightly transformed and shoved into the installer binary. Though it probably wouldn't really get far into CI if the required fields were actually wrong.

So overall, I'd lean towards tighter typing, but at the same time I don't think it's a pressing issue (and ideally eventually we stop hardcoding `meta.json` in the installer). So I wouldn't block on it either.

Reply:
> So let me make sure I understand:
> 
>     * `Azure` uses only `Image` and `URL`

looks like it, though I will admit we don't use this type for FCOS so I don't know if the `image:` that's getting populated in the metadata for RHCOS is an actual bootable image or just the file name of the object in object storage.

> 
>     * `IbmCloud` uses `Bucket`, `Object`, `Region`, and `Url`> 
>     * `PowerVirtualServer` uses `Bucket` and `Object`

IIUC both of these would populate `Bucket`, `Object`, `Region`, and `Url`. Obviously knowing enough about the structure of the cloud you can introspect URL from the other data. It's mostly there as a nice to have. 

> 
> I mean, it depends how "well-typed" we want to be. Those clearly could use separate types, and then validation would be that much stronger (which means e.g. we don't run the risk of pushing an `azure` key with fields which don't make sense). While in FCOS these details are mostly internal, because we have layers of metadata on top with very different schemas, in RHCOS the `meta.json` is only lightly transformed and shoved into the installer binary. Though it probably wouldn't really get far into CI if the required fields were actually wrong.
> 
> So overall, I'd lean towards tighter typing, but at the same time I don't think it's a pressing issue (and ideally eventually we stop hardcoding `meta.json` in the installer). So I wouldn't block on it either.

I honestly think we need to take a step back and make sure all of our metadata variants make more sense.

For now I propose we just go forward with what we have here.

Reply:
> For now I propose we just go forward with what we have here.

Yup sure, WFM!

Reply:
yes so IBMCloud artifacts and PowerVS artifacts would have the same fields as they both use the same cloud and similar object storage buckets.